src/compiler/factory.ts

```diff
@@ -1462,15 +1462,15 @@ namespace ts {
             : node;
     }
 
-    export function createTypeAliasDeclaration(name: string | Identifier, typeParameters: TypeParameterDeclaration[] | undefined, type: ts.TypeNode) {
+    export function createTypeAliasDeclaration(name: Identifier, typeParameters: TypeParameterDeclaration[] | undefined, type: TypeNode) {

```

Comment:
This should still support `string`.  The update method is the only one that only accepts `Identifier`